examples/widgets/label_texture_size.py

```diff
@@ -0,0 +1,182 @@
+authorities insisted on its being received, for good or for evil, in the
+superlative degree of comparison only.
+"""
+
+_kv_code = """
+# Note: StackLayout, ToggleButton, and HeadingLabel are defined at the bottom.
+Widget:
+    GridLayout:
+        id: grid_layout
+        cols: 2
+        top: root.top
+        width: root.width
+        height: cm(6)
+
+        HeadingLabel:
+            text: "GridLayout (1st column): Default, no text_size set"
+
+        HeadingLabel:
+            text: 'GridLayout: (2nd column): text_size bound to size'
+
+        Label:
+            id: left_content
+            canvas:
+                Color:
+                    rgb: 68/255.0, 164/255.0, 201/255.0
+                Line:
+                    rectangle: self.x, self.y, self.width, self.height
+
+        Label:
+            id: right_content
+            text_size: self.size
+            padding: dp(6), dp(6)
+
+            canvas:
+                Color:
+                    rgb: 68/255.0, 164/255.0, 201/255.0
+                Line:
+                    rectangle: self.x, self.y, self.width, self.height
+
+        StackLayout:
+
+            ToggleButton:
+                text: 'Disable'
+                on_state:
+                    left_content.disabled=self.state=='down'
+
+        StackLayout:
+
+            ToggleButton:
+                text: 'Shorten'
+                on_state:
+                    right_content.shorten=self.state=='down'
+
+            ToggleButton:
+                text: 'max_lines=3'
+                on_state:
+                    right_content.max_lines=3 if self.state=='down' else 0
+
+            Spinner:
+                text: 'bottom'
+                values: 'bottom', 'middle', 'top'
+                on_text: right_content.valign=self.text
+                size_hint: None, None
+                height: self.texture_size[1]
+                padding: dp(10), dp(8)
+
+    Label:
+        # This Label will not wrap because text_size is not specified.
+        id: title
+        text: 'A Tale of Two Cities, by Charles Dickens'
+        font_size: '20sp'
+        y: bottom_content.top
+        padding: mm(4), mm(4)
+        # Binding size to texture_size is still required for padding to work.
+        size: self.texture_size
+
+    Label:
+        id: bottom_content
+        size: self.texture_size
+        # This Label wraps because text_size width is set.
+        text_size: root.width, None
+        padding: mm(4), mm(4)
+
+        # Outline Label size
+        canvas:
+            Color:
+                rgb: 68/255.0, 164/255.0, 201/255.0
+            Line:
+                rectangle: self.x, self.y, self.width, self.height
+
+    Button:
+        # Button is a subclass of Label and can be sized to text in the same way
+        text: 'Reset'
+        padding: mm(6), mm(3)
+        size: self.texture_size
+        # Setting right this way didn't work, not sure why.
```

Comment:
This isn't actually a bug, you can't set right and then change width and expect right to stay aligned. I created #3430 and new documentation explains this. I will remove these comments.